modules/vmss/main.tf

```diff
@@ -2,6 +2,14 @@ locals {
   password = sensitive(var.authentication.password)
 }
 
+# https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/data-sources/public_ip_prefix
+data "azurerm_public_ip_prefix" "allocate" {
```

Comment:
That comment applies not only to this PR, but also to:
- #72 
- #71 
- #70 
- #68 
- #67 

Maybe instead of repeating code for `azurerm_public_ip_prefix` in multiple modules, it's worth to discuss possibility to create new module e.g. `public_ip`, in which we can manage (resources or data sources) for:
- public IPs
- public IP prefixes

Then we have idendependent module for managing public IPs and even if we destroy `VMSS` or `VM-Series` instances by removing all module `vmss` or `vmseries` invocations, then we are not loosing public IPs , still we can recreated VM-Series with the same IP only by replacing instance of `vmseries` module.

Reply:
Hmm, but you would have to call the `public_ip` module within other modules like `vmss` for instance, wouldn't you? Similarly to what we do inside the `test_infrastructure` module. I think it wouldn't change the behaviour of `terraform destroy` command.

Reply:
Module `public_ip` for IPs an prefixes I propose to call outside `vmss` module - we can do it in glue code and put as an input ID of the public IP (or prefix) to `vmss` module.

Reply:
Got it, but it's a little bit tricky because different resources differ on how the public IPs are created. In `vmseries` module for instance we create a PIP as a separate TF resource (and that's a good use case for a `public_ip` module I guess) but in `vmss` module, PIP is created as a part of the scale set TF resource. But I guess we could still just create/source a public IP prefix for the PIP created within the `vmss` module. Let's discuss this with Migara.

Reply:
We can discuss with Migara or you can go with your approach - I approved the PR. Thansk